src/tests/ftest/cart/util/daos_sys_logscan.py

```diff
@@ -0,0 +1,293 @@
+                            #print(f"PROCESS FOUND pool {puuid} BEGIN term {term}: maps dictionary exists.")
+                            #if (self._pools[old_term]["maps"]) != {}:
+                                #print(f'PROCESS FOUND pool {puuid} BEGIN term {term}: maps dictionary is non-empty, with {len(list(self._pools[puuid][old_term]["maps"].keys()))} keys')
+                        if self._pools and self._pools[puuid][old_term]["maps"] != {}:
+                            last_mapver = list(self._pools[puuid][old_term]["maps"].keys())[-1]
+                            pmap_versions = self._pools[puuid][old_term]["maps"][last_mapver]
+                            pmap_versions["carryover"] = True
+                            #pmap_versions["rebuild_gens"] = {}
+                    else:
+                        pmap_versions = {}
+                    self._pools[puuid][term] = {"rank": rank, "begin_time": datetime, "end_time": "", "host": host, "pid": pid, "logfile": fname, "maps": pmap_versions}
+                    #print(f"{datetime} FOUND pool {puuid} BEGIN\tterm {term} pmap_versions empty: {str(pmap_versions == {})} rank {rank}\t{host}\tPID {pid}\t{fname}")
+                    continue
+
+                # Find pool term end (PS leader step_down)
+                match = self.re_step_down.match(msg)
+                if match:
+                    puuid = match.group(1)
+                    term = int(match.group(2))
+                    if (term != self._cur_term[puuid]):
+                        print(f"WARN: step_down term={term} != cur_term={self._cur_term}")
+                    self._cur_ldr_rank[puuid] = -1
+                    self._cur_term[puuid] = -1
+                    self._pools[puuid][term]["end_time"] = datetime
+                    #print(f"{datetime} FOUND pool {puuid} END\tterm {term} rank {rank}\t{host}\tPID {pid}\t{fname}")
+                    continue
+
+                # TODO: find target status updates (precursor to pool map updates)
+
+                # Find pool map updates
+                # FIXME: but only on the current PS leader engine / term? And carry over latest pmap version from prior term to start with?
+                match = self.re_pmap_update.match(msg)
+                if match:
+                    puuid = match.group(1)
+                    from_ver = int(match.group(2))
+                    to_ver = int(match.group(3))
+                    # ignore if this engine is not the leader
+                    if puuid not in self._pools or rank != self._cur_ldr_rank[puuid]:
+                        continue
+                    term = self._cur_term[puuid]
+                    self._pools[puuid][term]["maps"][to_ver] = {"carryover": False, "from_ver": from_ver, "time": datetime, "rebuild_gens": {}}
+                    #print(f"FOUND pool {puuid} map update {from_ver}->{to_ver} rank {rank}\t{host}\tPID {pid}\t{fname}")
+                    continue
```

Comment:
This function has a lot of these of the format
```
match = ...
if match:
    ...
```

I think it would be easier to manage if each was a helper function. Then this function could either loop over each one, or call out to each function. It's generally not pythonic to have large functions.

There are many ways to do this, but some general examples might be (depending on what you need)
```
def fun1(arg):
    print(arg)

def fun2(arg):
    print(arg)

all_funs = (fun1, fun2)
for _fun in all_funs:
    _fun('test')
```

```
def fun1(arg1, arg2):
    print(arg1, arg2)

def fun2(arg1, arg2):
    print(arg1, arg2)

all_funs_args = (
    (fun1, ('arg1', 'arg2')),
    (fun2, ('arg1', 'arg2')))

for _fun, args in all_funs_args:
    _fun(*args)
```